### PR TITLE
Feat: create read job post

### PR DESCRIPTION
**Pull request by uloamaka**
​The Create and Read functionalities for a job post were implemented, The Create has an endpoint 'CreateJobPost', The Read has two endpoint 'FetchAllJobPost' and 'FetchJobPostByID'.
## Description

Implemented the Create functionality and also the Read functionalities for the job post/listing feature, these endpoint followed the requirement of the code base such that the router -> controller -> service -> model -> repository ... and hence no direct interaction with db from controllers or service file, the controller only handles validation, calling of service and logging appropriate messages.

​
## Related Issue (Link to Github issue)

[54](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/54)
[53](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/53)
[52](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/52)
​
## How Has This Been Tested?

The test case covers create a new job post, and fetching the created job post, and also all job post available in the DB.
​E2E Test case can be run by:  go test -v .\tests\jobpost_test.go
## Screenshots (if appropriate - Postman, etc):
CREATE ENPOINT:
Screenshot (282)
FETCH JOB POST BY ID: 
Screenshot (283)
FETCH ALL JOB POST:

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/controller/jobpost/jobpost.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+type Controller struct {
+  Db        *storage.Database
+  Validator *validator.Validate
+  Logger    *utility.Logger
+  ExtReq    request.ExternalRequest 
+}
+
+func (base *Controller) CreateJobPost(c *gin.Context) {
+	var req models.JobPost
```
Comment (Cyberguru1) [inline pkg/controller/jobpost/jobpost.go:25]: create a request model for this and specify all required fields, check the other models for reference

File: services/jobpost/jobpost.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"fmt"
+
+	"github.com/hngprojects/hng_boilerplate_golang_web/internal/models"
+	"github.com/hngprojects/hng_boilerplate_golang_web/utility"
+	"gorm.io/gorm"
+)
+
+func ValidateCreateJobPost(req models.JobPost) error {
```
Comment (Cyberguru1) [inline services/jobpost/jobpost.go:11]: if there are unique fields, validate if the job exists in db here, as this is a redundant function if you do the first review, check other services for reference

**Review by urizennnn — DISMISSED** [review]

File: services/jobpost/jobpost.go (hunk 2 above)
Comment (uloamaka) [reply services/jobpost/jobpost.go:11]: for this job post there are no current unique fields.

File: pkg/controller/jobpost/jobpost.go (hunk 1 above)
Comment (Cyberguru1) [reply pkg/controller/jobpost/jobpost.go:25]: still waiting for this to be done

**Pull request closed without merging**